_docs/cardano/2017-01-01-proof-of-stake.md

```diff
@@ -9,83 +9,173 @@ visible: true
-by others, make a block of those transactions, sign it with your secret key and
-publish it to the network. Of course, you don't have to do it manually — your
-node will take care of everything.
-
-## Multi Party Computation
-
-The matter of fueling Follow the Satoshi with randomness is another problem in
-itself. We are using Multi Party Computation approach: select nodes provide
-the so-called “commitments”, and then those get “revealed”, producing a random
-value generated independently by participants of the network.
-
-<!-- TODO PVSS -->
+value in the Cardano SL system”. Please read about [Balance and Stake in Cardano
+SL](/cardano/balance-and-stake/) for more details.
+
+## Slot Leaders
+
+Nodes with a positive stake are called stakeholders, and only stakeholders may
+participate in running the protocol. Moreover, to be able to generate new blocks
+for the blockchain, stakeholder must be elected as a slot leader. Slot leader may
+listen to transactions announced by other nodes, make a block of those transactions,
+sign this block with its secret key and publish it to the network.
+
+You can think of a slot leader as a miner in Bitcoin, but consensus mentioned above
+defines who will be able to mine, when and how much.
+
+## Epochs and Slots
+
+Ouroboros protocol divides the physical time into **epochs**, and each epoch is
+divided into **slots**:
+
+```
++----------+----------+-------+----------+--------------------> t
+|  slot 0  |  slot 1  |  ...  |  slot N  |
+
+ \                                      / \
+  -------------- epoch M ---------------   -- epoch M+1 -- ...
+```
+
+Please note that slot is relatively short period of time (for example, 20 seconds).
+
+Each slot has one and only one leader (slot leader, SL):
+
+```
++----------+----------+-------+----------+----> t
+|  slot 0  |  slot 1  |  ...  |  slot N  |
+
+    SL 0       SL 1               SL N
+```
+
+Slot leader has a (sole) right to produce one and only one block during his slot:
+
+```
+  +------+   +------+           +------+
+  | Bl 0 |<--| Bl 1 |<-- ... <--| Bl N |
+  +------+   +------+           +------+
++----------+----------+-------+----------+----> t
+|  slot 0  |  slot 1  |  ...  |  slot N  |
+
+    SL 0       SL 1               SL N
+```
+
+It means that the number of slot leaders is strictly equal to the number of slots
+in epoch (let's call this number `N`), so it is impossible to produce more than `N`
+blocks during an epoch.
+
+If slot leader missed his slot (for example, he was offline at that moment), he
+lost his right to produce a block until he will be elected again. So one or more
+slots can remain empty (i.e. without generated blocks), but please note that the
+most of blocks (i.e. at least 50% + 1) **must** be generated during an epoch.
+
+## Slot Leaders Election
+
+Slot leaders are elected from among stakeholders. Please note that not all stakeholders
+participate in this election, but only ones who has enough stake (for example, 2% of
```

Comment:
I guess it can be interpreted incorrectly by potential reader. One possible interpretation is that only stakeholders with at least 2% stake can become slot leaders. But it's not true, even someone with 1 ADA can become a slot leader. However, only stakeholders with at least 2% stake can affect leader selection process.

Reply:
Upd: I see that later you say that everyone can be a slot leader, but I still think this sentence may lead to some confusion. It's subjective though, maybe it's just me who thinks so.